src/util/value_history.rs

```diff
@@ -64,17 +64,17 @@ mod test {
 
         let mut h = ValueHistory::new(11, Duration::from_secs(1));
 
-        h.push(now - Duration::from_millis(1500), 22);
-        h.push(now - Duration::from_millis(500), 22);
+        h.push(now.checked_sub(Duration::from_millis(1500)).unwrap(), 22);
+        h.push(now.checked_sub(Duration::from_millis(500)).unwrap(), 22);
```

Comment:
These `-` are deliberate. If we are using `checked_sub` we need to motivate why. I need to write a line about it, cause I only discussed it internally.

This is from our internal Slack:

> As an overall observation of the code going into str0m. I'm seeing an increased use of "saturating_sub" and "saturating_duration". Not saying there is a problem (I should review it). Generally `-` is vastly simpler to read for an observer. Whenenver we use "saturating_xxx" I'd like to see a code comment why the use is correct.
>
> Like: is underflow expected at this call site, if yes, use saturating_sub (and code comment why) 
> Is it not expected? Normal sub and panic if it's an incorrect state.

Reply:
fixed

Reply:
There's a good reason here, see https://rust-lang.github.io/rust-clippy/master/index.html#unchecked_duration_subtraction and https://github.com/rust-lang/rust-clippy/pull/9570.

On some systems this can panic if you run it early enough in the process lifecycle because of how Instant is represented internally. If this where to actually panic we'd fix it by using `checked_sub.unwrap_or(now)` or something like that, so I think we can do it immediately. Generally, all panics of `Instant - Duration` will not be useful to us and should use the above pattern I think.

Reply:
These are tests. If we use `checked_sub()` and replace it with `now` (or saturating), the tests will fail instead. The complexity introduced of not using `-` here is not worth it.

Reply:
Sure, for the tests I don't disagree. Elsewhere though

Reply:
I don't want to generalize and say "we should always do this",  but look at case by case. 

My gut feeling is that most instances the logic would break. I prefer panics to incorrect state.